lib/sober_swag/reporting/output/defer.rb

```diff
@@ -0,0 +1,49 @@
+module SoberSwag
+  module Reporting
+    module Output
+      ##
+      # Defer loading of an output for mutual recursion and/or loading time speed.
+      # Probably just do this for mutual recursion though.
+      #
+      # Note: this *does not* save you from infinite schema generation.
+      # This type *must* return some sort of {Referenced} type in order to do that!
+      class Defer < Base
```

Comment:
nit: It would be nice to have some examples for how end users are supposed to actually use this. Is it automatically invoked on self-references or does the person writing the schema need to use it in the type declaration?

Reply:
9709025

Reply:
Very nice